packages/app-mining/src/components/WithdrawModal.tsx

```diff
@@ -62,9 +63,10 @@ const WithdrawModal = (props: StakePoolModalProps): JSX.Element => {
     if (!userStakeInfo || !decimals) return
     const yourDelegation = userStakeInfo.shares
       .mul(stakePool.totalStake.div(stakePool.totalShares))
+      .floor()
       .div(decimals)
-      .toString()
```

Comment:
Either `floor()` or `toFixed()` is needed, they do the same thing actually. When a floored int is divided by `decimals`, its number of decimal places will not exceed 12.